src/components/csvUploads/DisplayCSV.tsx

```diff
@@ -0,0 +1,357 @@
+  const [logs, setLogs] = useState([]);
+
+  const indexOfLastPatient = currentPage * dataPerPage;
+  const indexOfFirstPatients = indexOfLastPatient - dataPerPage;
+  const currentData =
+    data.length > 0 && data?.slice(indexOfFirstPatients, indexOfLastPatient);
+
+  const paginate = (pageNumber: number) => setCurrentPage(pageNumber);
+  const { user } = storage.loadData();
+
+  useEffect(() => {
+    const fetchCsvFiles = async () => {
+      const res = await getCsvFiles(user.display);
+      if (res) {
+        const response = await res.json();
+        setData(response);
+      } else {
+        console.log("Unable to fetch csv files");
+      }
+    };
+
+    fetchCsvFiles();
+  }, []);
+
+  interface CsvFile {
+    file_name: string;
+    upload_time: Date;
+    status: string;
+    total_records: number;
+    existing_records: number;
+    failed: number;
+    successful: number;
+    file_type: string;
+    id: number;
+    eid_file_upload_metadata_id: number;
+  }
+
+  function ModalPop() {
+    function closeModal() {
+      setIsOpen(false);
+    }
+
+    return (
+      <div>
+        <Modal
+          isOpen={modalIsOpen}
+          onRequestClose={closeModal}
+          style={customStyles}
+        >
+          <div className="relative w-full">
+            <button
+              className="absolute right-0 top-0 text-2xl"
+              onClick={closeModal}
+            >
+              <AiOutlineClose />
+            </button>
+            <div className="w-full p-4">
+              <h1 className="text-xl font-bold text-red-600 underline">
+                Error Logs
+              </h1>
+              <ul className="p-4">
+                {logs &&
+                  logs.map((log: any, idx) => {
+                    const parsedLog = JSON.parse(log);
+                    return (
+                      <li
+                        key={idx}
+                        className="text-sm text-gray-600 border-spacing-2 p-2"
+                      >
+                        <span>{idx + 1}. </span>
+                        {parsedLog.message}
+                      </li>
+                    );
+                  })}
+              </ul>
+            </div>
+          </div>
+        </Modal>
+      </div>
+    );
+  }
+
+  const handleVoidedCsv = async (id: number) => {
+    const voidedCsv = data.filter((item: CsvFile) => item.id !== id);
+    setData(voidedCsv);
+    swal({
+      title: "Are you sure?",
+      text: "Once deleted, you will not be able to recover this item!",
+      icon: "warning",
+      buttons: ["No", "Yes"],
+      dangerMode: true,
+    }).then(async (willDelete) => {
+      if (willDelete) {
+        try {
+          const res = await deleteCsvFile(id);
+          const response = await res.json();
+          if (response.affectedRows === 1) {
+            toast.success("File deleted successfully");
+            setTimeout(() => {
+              window.location.reload();
+            }, 6000);
```

Comment:
Why the timeout? 6s??
`async` - `await` should be sufficient in this case.

Reply:
:+1: